tests/test_util.py

```diff
@@ -105,3 +107,68 @@ def test_clean_stale_shared_memory():
     # If clean up is successful, it should raise FileNotFoundError Exception
     with pytest.raises(FileNotFoundError):
         _ = BuiltinSharedMemory(name, False, 64)
+
+
+@pytest.mark.parametrize('folder_urls', [
+    'local_accessible', 'remote', 'local_unaccessible', 'tuple_local_accessible',
+    'tuple_partial_local_accessible'
+])
+@pytest.mark.parametrize('out', ['local_str', 'remote_str', 'tuple'])
+@pytest.mark.usefixtures('local_remote_dir')
+@pytest.mark.parametrize('keep_local', [True, False])
+def test_merge_index(local_remote_dir: Tuple[str, str], keep_local: Any, folder_urls: Any,
```

Comment:
Suggestion: To test the method functionality, I would not worried about cloud provider URLs. Just test whether merge index.json happens correctly or not using local directory. For example:

Test: `folder_urls` is just a `local_dir`, create two to three sub-dataset and apple `merge_index` method. Validate the number of shard files of the final merged index.json. For validation number of shard files, you can get the `basename` field in the `set` and count the length of `set`.